include/ginkgo/core/base/matrix_data.hpp

```diff
@@ -190,6 +191,7 @@ struct matrix_data {
     matrix_data(dim<2> size_, RandomDistribution&& dist, RandomEngine&& engine)
         : size{size_}
     {
+        nonzeros.reserve(size[0] * size[1]);
```

Comment:
reserving the dense size is quite easy to exceed the memory limit

Reply:
this is a constructor for a dense random matrix, so that is intended.